connectors/sources/sharepoint_server.py

```diff
@@ -668,6 +648,33 @@ async def close(self):
         """Closes unclosed client session"""
         await self.sharepoint_client.close_session()
 
+    async def _remote_validation(self):
+        """Validate configured collections
+        Raises:
+            ConfigurableFieldValueError: Unavailable services error.
+        """
+
+        for collection in self.sharepoint_client.site_collections:
+            is_invalid = True
+            async for _ in self.sharepoint_client._api_call(
+                url_name=PING,
+                site_collections=collection,
+                host_url=self.sharepoint_client.host_url,
+            ):
+                is_invalid = False
+            if is_invalid:
+                self.invalid_collections.append(collection)
+        if self.invalid_collections:
+            raise ConfigurableFieldValueError(
+                f"Collections {', '.join(self.invalid_collections)} are not available."
+            )
```

Comment:
Can you fetch all collections in one go and then compare?

E.g.:

```
user_defined_collections = self.config['site_collections']
site_collections_on_server = await self.sharepoint_client.get_all_site_collections()

# Then compare user_defined_collections against site_collections_on_server by name

```

Reply:
There is no such api available which returns list of all collections.